### ISSUE TITLE
Screen reader support

### ISSUE THREAD

**Issue by DanielOaks** [sekaiproject/ponscripter-fork#79]
Hopefully we can add okay screen-reader support sometime down the line (for people who are visually impared, for instance). Schwarzstorch has seemed to do it in [this thread](http://steamcommunity.com/app/264380/discussions/0/540739861382548810/), and if we can get his code or just do it ourselves, it would be nice to be able to add that feature.

This isn't a definite or a priority issue, but it's a nice-to-have, if we can include it without too much hassle.

**edit:**
Found Schwarzstorch's branch and his changes [here](https://github.com/Schwarzstorch/ponscripter-fork/compare/accessibility). It's good, but I think I can take the same approach, clean it up a bit, and have something that's a little nicer for the screen readers.

Work going on in the [screen-readers](https://github.com/sekaiproject/ponscripter-fork/tree/screen-readers) branch.

**Comment by DanielOaks** [comment]
Got another updated source archive from @Schwarzstorch , which is integrated into the project now (except for the actual Tolk library which makes it all workable on Windows).

The problem with linking to the [Tolk library](https://github.com/dkager/tolk) is that that's an LGPLv3 project, and this is GPLv2-or-later project (with a dependency on the LGPLv2-only SMPEG library).

License talk is really, /really/ boring but I don't want to change the license like that, since it would mess with SMPEG as well as cut the other *Nscripter projects from taking code from us (unless we say that it only becomes GPLv3 when we make an accessibility build on Windows, which is irritating and annoying).

So... I'll keep having more of a look. It seems the [other project](https://github.com/qtnc/UniversalSpeech) linked on the Tolk page is also GPLv3.

It might make more sense for us to bind directly to the SAPI library? Or if there is actually another unencumbered (or GPLv3-or-later encumbered) library, we could use that directly.

**Comment by Schwarzstorch** [comment]
OMG. It seems we need a new interface for screenreaders.

**Comment by DanielOaks** [comment]
Yeah, not much we can do unfortunately. I'll try to fire up my Windows VM and get the SAPI code working on there over the next few days, then see how Linux wants us to do it.

**Comment by Schwarzstorch** [comment]
You know, before I started using Tolk, I wrote (reinvented the wheel) an interface for NVDA and Jaws (that worked). I'm thinking to make something similar to Tolk (fortunately, all those screen readers are using pretty much the same algorithm).

**Comment by DanielOaks** [comment]
Fair enough. I've been thinking of something similar (though if I were
writing one, I'd go either MIT/BSD/public domain licensing, rather than
GPL). Been meaning to add this sorta simple functionality to a few
different projects, just always been stopped by the cross-platform stuff,
and licensing issues with existing libraries.

**Comment by Schwarzstorch** [comment]
<a href="http://www.mediafire.com/download/w1a2y2u3xqp5zdw/ScreenReader.zip">Here</a> you go.
I've tested it on jaws, nvda, system access, supernova, zoomtext, window eyes and sapi (all lastest versions, except for jaws (v.15)). Sapi is working on XP 32 and 8.1 64.
No idea how to test braille... brltty doesn't seem to work properly on windows or just nvda isn't working with it properly.
I think, we can do nothing about their sdks and libs; if they allows us to distribute their stuff with our apps, then we're good to go. Another problem is to find those sdks, libs and docs.

**Comment by DanielOaks** [comment]
Awesome, thanks!

I'll be having a look through and seeing how to integrate this nicely.
Thanks a bunch for the code, though.

Just curious, what license are you putting this code under? And is it
alright if I made a git repo for it, or do you want to create that yourself?

**Comment by Schwarzstorch** [comment]
I don't really care about type of the license. Do what you think is better for you (git, licensing, whatever).

I've managed to make brltty to work with nvda, but the output was like - here's a bunch of text and do whatever you want with it. It's not like we should throw text at people and hope for the best...

**Comment by Schwarzstorch** [comment]
Updated some things. Not sure on, should I pull the request or not. Take a look, let me know what you think.
